### DISCUSSION TITLE
Should `write_xlsx()` return `file` invisibly?

### DISCUSSION THREAD

**Discussion by olivroy** [JanMarvin/openxlsx2#738] [Ideas]
Similar to many `write_` functions in R, should `write_xlsx()` return `file` invisbly?

This is an unlikely workflow imo, but it currently works

```r
write_xlsx(mtcars, file = temp_xlsx())$add_worksheet("test")$open()
```

Edit: I tend to use a lot workflows like
```r
readr::write_csv(mtcars, file = "file.csv")
fs::file_show(.Last.value)
```

Edit2: I guess you may do `write_xlsx(mtcars, file = "file.xlsx")$open()`, but it is openxlsx2-specific.

**Comment by JanMarvin** [comment]
Don't really have an opinion. You can use `$path()` iirc

**Reply by JanMarvin** [reply]
Moved this to a discussion. Unless for testing I don't use `write_xlsx()`, hence personally I don't really care. Afaik we simply inherited this from `openxlsx`. But it behaves like this for quite some time already. And unless some more people want this, I don't think it's worth bricking things just to behave like some other package.
